onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -1265,6 +1265,35 @@ def _infer_Transpose(self, node):
                                                             axes=tuple(perm)).flatten().tolist()
 
     def _infer_Unsqueeze(self, node):
+        input_shape = self._get_shape(node, 0)
+        op_set = get_opset(self.out_mp_)
+
+        if op_set < 13:
+            axes = get_attribute(node, 'axes')
+            assert self._try_get_value(node, 1) is None
+        else:
+            # In opset version 13, 'axes' of Unsqueeze are provided in the second input instead of attribute
+            axes = self._try_get_value(node, 1)
+            assert get_attribute(node, 'axes') is None
+
+        assert axes is not None, 'axes is required for Unsqueeze'
+
+        output_rank = len(input_shape) + len(axes)
+        axes = [handle_negative_axis(a, output_rank) for a in axes]
+        assert len(axes) == len(set(axes)), "duplicated axes is not allowed for Unsqueeze"
+
+        output_shape = [1] * output_rank
+        j = 0
+        for i in range(output_rank):
+            if i not in axes:
+                output_shape[i] = input_shape[j]
+                j += 1
+
+        vi = self.known_vi_[node.output[0]]
+        vi.CopyFrom(
+            helper.make_tensor_value_info(node.output[0], self.known_vi_[node.input[0]].type.tensor_type.elem_type,
+                                          output_shape))
```

Comment:
seems a patch to onnx shape inference. I'm OK with the change in symbolic_shape_infer, and it would be better if we also open a bug in onnx for it.

Reply:
onnx shape inference could handle the test model.

After some investigation,  the root cause is initializer value of axes is not passed to onnx shape inference (only the shape of axes is passed which is not enough):
https://github.com/microsoft/onnxruntime/blob/418623355ae263664e7440a72dbc3b72dcd24caf/onnxruntime/python/tools/symbolic_shape_infer.py#L333-L336

I think an alternative fix is to pass initializer value that might impact shape inference. That could be future work.